File: JNA/Java/tests/prng/performance test/prng.java
public class prng {
	public static void main(String[] args) {
		long startTime = System.currentTimeMillis();
		ArcanaJNI aJNI = new ArcanaJNI();

		if (aJNI.ktb_init() == ArcanaJNI.ktb_errno.KTB_ERR_NO_ERROR)
			System.out.println(Utils.getCurrentTime() + "ktb_init() ok");
		else {
			System.out.println(Utils.getCurrentTime()
					+ "Error while initializating ktb_init()");
			System.exit(1);
		}

		if (!prng.exemple1(aJNI)) {
			aJNI.ktb_clear();
			System.exit(1);
		}

		aJNI.ktb_clear();
		long endTime = System.currentTimeMillis() - startTime;
		System.out.println(endTime + "ms");
	}

	public static boolean exemple1(ArcanaJNI aJNI) {

		int buffer_size = 50;
		byte[] buffer = new byte[buffer_size];
		if (aJNI.ktb_prng_fill_buffer(0, buffer) == ArcanaJNI.ktb_errno.KTB_ERR_NO_ERROR)
			System.out.println(Utils.getCurrentTime() + "ktb_fill_buffer() ok");
		else {
			System.out.println(Utils.getCurrentTime()
					+ "Error while initializating ktb_fill_buffer()");
			Utils.freeBuffer(buffer);
			return false;
		}

		Utils.printBuffer(buffer);

		Utils.freeBuffer(buffer);
		return true;
	}
}

File: JNA/Java/tests/prng/performance test/prng.c
#include <stdio.h>
#include <time.h>
#include <stdint.h>
#include <ktb.h>
#include <ktb/hash.h>

int example1() {
   int buffer_size = 50;
   uint8_t *buffer = malloc(buffer_size);

	if(ktb_prng_fill_buffer(NULL, buffer, buffer_size) == KTB_ERR_NO_ERROR)
		printf("ktb_prng_fill_buffer() ok\n");

   int i;
   for(i = 0; i < buffer_size; ++i)
      printf("%02X ", buffer[i]);
   printf("\n");

   for(i = 0; i < buffer_size; ++i)
      buffer[i] = 0;
   free(buffer);

	return 1;
}

int main(void) {

	struct timespec tsi, tsf;

	clock_gettime(CLOCK_MONOTONIC, &tsi);
	if(ktb_init() == KTB_ERR_NO_ERROR) 
		printf("ktb_init() ok\n");

	if(!example1()) {
		ktb_clear();
		exit(1);
	}

	ktb_clear();

	clock_gettime(CLOCK_MONOTONIC, &tsf);
	double elaps_s = difftime(tsf.tv_sec, tsi.tv_sec);
	long elaps_ns = tsf.tv_nsec - tsi.tv_nsec;
	printf("Elapsed time : %lf ms\n", elaps_s + ((double)elaps_ns) / 1.0e6);
	return 0;
}


